pkg/tasks/alertmanager.go

```diff
@@ -77,7 +77,7 @@ func (t *AlertmanagerTask) create(ctx context.Context) error {
 		return fmt.Errorf("initializing Alertmanager configuration Secret failed: %w", err)
 	}
 
-	err = t.client.CreateIfNotExistSecret(ctx, s)
+	err = t.client.CreateOrUpdateSecret(ctx, s)
```

Comment:
Good catch for the rbac proxy secrets, we should for sure update them when needed. Not sure about how we can easily have a regression test for this.

For some secrets like the one containing alertmanager config, CMO shouldn't override it as it may be customized by users https://docs.openshift.com/container-platform/4.16/observability/monitoring/managing-alerts.html#configuring-notifications-for-default-platform-alerts_managing-alerts

Reply:
Yeah, that's a tricky case, since if we make updates to that secret between releases, and upgraded CMO version won't reconcile it.

Reply:
I'll revert the change for those cases.

Reply:
we can go through them one by one ;)
But AFAIK, users aren't meant to change the rbac proxy secrets.

Reply:
Refactored to exclude `alertmanager` configuration secrets for UWM and in-cluster environments, as users may want to modify those.

Reply:
I have an idea of an e2e test for this:
- keep a list of secretsNotToSync (with the alertmanager config secret etc, the first test run will help us build it)
- loop over all the secrets in the two monitoring namespaces (if we can get all the operands enabled it'd be great) if not in secretsNotToSync, alter them (add a `foo: bar` data entry or sth)
- trigger a CMO sync and make sure the added entry gets deleted from all secrets.

We can extend this to configmaps, RBAC etc. in the future.
If we can have it for secrets as part of this PR it'd be great.